documentation/test/test_process_provider.py

```diff
@@ -0,0 +1,57 @@
+import pytest
+import local_gen
+
+from unittest.mock import mock_open, patch, MagicMock
+from pathlib import Path
+
+class TestProcessProvider:
+    @pytest.fixture
+    def test_path(self):
+        return Path("/")
+    
+    @pytest.fixture
+    def test_string(self):
+        return "aws"
+
+    def test_provider_is_not_directory(self, test_path):
+        with patch.object(Path, 'is_dir', return_value=False) as mock_is_dir:
+            result = local_gen.process_provider(test_path, test_path, test_path, test_path, 0, [], {})
+            assert result == 0
+            mock_is_dir.assert_called_once()
+    
+    def test_metadata_failure(self, test_path):
+        mock_metadata_data = """{ 
+        no_data
+        }""" # Not parsable data
```

Comment:
nit: Inline comment should be separated by two-spaces (or more, for example for alignment) like [pep 8 recommends](https://peps.python.org/pep-0008/#inline-comments)
```suggestion
        }"""  # Not parsable data
```